Tasks/VsTest/distributedtest.ts

```diff
@@ -59,68 +60,169 @@ export class DistributedTest {
+        }
+        if (this.dtaTestConfig.numberOfTestCasesPerSlice) {
+            inputDataContract['DistributionSettings']['NumberOfTestCasesPerSlice'] = this.dtaTestConfig.numberOfTestCasesPerSlice;
+        }
+
+        if (this.dtaTestConfig.rerunFailedTests) {
+            inputDataContract['ExecutionSettings']['RerunSettings']['RerunFailedTests'] = true;
+            tl.debug("Type of rerun: " + this.dtaTestConfig.rerunType);
+            if (this.dtaTestConfig.rerunType === 'basedOnTestFailureCount') {
+                inputDataContract['ExecutionSettings']['RerunSettings']['RerunFailedTestCasesMaxLimit'] = this.dtaTestConfig.rerunFailedTestCasesMaxLimit;
+            } else {
+                inputDataContract['ExecutionSettings']['RerunSettings']['RerunFailedThreshold'] = this.dtaTestConfig.rerunFailedThreshold;
+            }
+            inputDataContract['ExecutionSettings']['RerunSettings']['RerunMaxAttempts'] = this.dtaTestConfig.rerunMaxAttempts;
+        }
 
         // If we are setting the path version is not needed
         const exelocation = path.dirname(this.dtaTestConfig.vsTestVersionDetails.vstestExeLocation);
         tl.debug('Adding env var DTA.TestWindow.Path = ' + exelocation);
 
-        // Split the TestWindow path out of full path - if we can't find it, will assume
-        // that this is nuget/xcopyable package where the dlls are present in test window folder
         const testWindowRelativeDir = 'CommonExtensions\\Microsoft\\TestWindow';
         if (exelocation && exelocation.indexOf(testWindowRelativeDir) !== -1) {
             const ideLocation = exelocation.split(testWindowRelativeDir)[0];
             tl.debug('Adding env var DTA.VisualStudio.Path = ' + ideLocation);
-            utils.Helper.addToProcessEnvVars(envVars, 'DTA.VisualStudio.Path', ideLocation);
+            inputDataContract['VisualStudioPath'] = ideLocation;
         } else {
-            utils.Helper.addToProcessEnvVars(envVars, 'DTA.VisualStudio.Path', exelocation);
+            inputDataContract['VisualStudioPath'] = exelocation;
         }
-        utils.Helper.addToProcessEnvVars(envVars, 'DTA.TestWindow.Path', exelocation);
+        inputDataContract['TestWindowPath'] = exelocation;
+        inputDataContract['TeamProject'] = tl.getVariable('System.TeamProject');
 
-        tl.debug('Total env vars set is ' + Object.keys(envVars).length);
+        // Pass the acess token as an environment variable for security purposes
+        let envVars: { [key: string]: string; } = <{ [key: string]: string; }>{};
+        utils.Helper.addToProcessEnvVars(envVars, 'DTA.AccessToken', this.dtaTestConfig.dtaEnvironment.patToken);
+
+        // Invoke DtaExecutionHost with the input json file
+        const inputFilePath = path.join(tl.getVariable('temp'), 'input.json');
+        DistributedTest.removeEmptyNodes(inputDataContract);
+        writeFileSync(inputFilePath, JSON.stringify(inputDataContract));
```

Comment:
file exceptions

Reply:
will add handlers

Reply:
actually no specific action is required here, there is a higher level catch block that prints out the error. And i checked the other file helper functions we have they just chain back the error and don't handle them in any way. Even if i did catch the only course of action is to fail the task and print error message which is done by the already enclosing catch block for this.